.github/workflows/default_workflow.yml

```
@@ -53,6 +63,36 @@ jobs:
     - name: Install dependencies
       run: pnpm install
 
+    - name: Initialize the Nx Cloud distributed CI run and stop agents when the build tasks are done
+      run: pnpm nx-cloud start-ci-run --distribute-on="manual"
+
+    # - name: Run commands in parallel
+    #   run: |
+    #     # initialize an array to store process IDs (PIDs)
+    #     pids=()
+
+    #     # function to run commands and store the PID
+    #     function run_command() {
+    #       local command=$1
+    #       $command &  # run the command in the background
+    #       pids+=($!)  # store the PID of the background process
+    #     }
+
+    #     # list of commands to be run on main has env flag NX_CLOUD_DISTRIBUTED_EXECUTION set to false
+    #     # run_command "NX_CLOUD_DISTRIBUTED_EXECUTION=false pnpm nx-cloud record -- nx format:check"
+
+    #     # list of commands to be run on agents
+    #     run_command "pnpm nx run-many --targets test,e2e --exclude devextreme devextreme-themebuilder devextreme-angular devextreme-react devextreme-vue devextreme-react-storybook devextreme-main devextreme-angular-playground devextreme-demos devextreme-react-playground devextreme-vue-playground"
+
+    #     # wait for all background processes to finish
+    #     for pid in ${pids[*]}; do
+    #       if ! wait $pid; then
+    #         exit 1  # exit with an error status if any process fails
+    #       fi
+    #     done
+
+    #     exit 0 # exits with success status if a all processes complete successfully
+
     - name: Run targets
       run: >
         pnpx nx run-many
```

```suggestion
        pnpm nx run-many
```
nx is already installed at this point